src/Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.cpp

```diff
@@ -0,0 +1,140 @@
+#include "Common/ErrorCodes.h"
+#include "Common/logger_useful.h"
+#include "Columns/ColumnNullable.h"
+#include "Columns/ColumnsDateTime.h"
+#include "DataTypes/DataTypeNullable.h"
+
+
+#include "Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.h"
+
+
+namespace DB::ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+}
+
+using namespace DB;
+
+namespace Iceberg
+{
+
+Iceberg::PartitionTransform getTransform(const String & transform_name)
+{
+    if (transform_name == "year" || transform_name == "years")
+    {
+        return Iceberg::PartitionTransform::Year;
+    }
+    else if (transform_name == "month" || transform_name == "months")
+    {
+        return Iceberg::PartitionTransform::Month;
+    }
+    else if (transform_name == "day" || transform_name == "date" || transform_name == "days" || transform_name == "dates")
+    {
+        return Iceberg::PartitionTransform::Day;
+    }
+    else if (transform_name == "hour" || transform_name == "hours")
+    {
+        return Iceberg::PartitionTransform::Hour;
+    }
+    else if (transform_name == "identity")
+    {
+        return Iceberg::PartitionTransform::Identity;
+    }
+    else if (transform_name == "void")
+    {
+        return Iceberg::PartitionTransform::Void;
+    }
+    else
+    {
+        return Iceberg::PartitionTransform::Unsupported;
+    }
+}
+
+DateLUTImpl::Values getValues(Int32 value, Iceberg::PartitionTransform transform)
+{
+    if (transform == Iceberg::PartitionTransform::Year)
+    {
+        return DateLUT::instance().lutIndexByYearSinceEpochStartsZeroIndexing(value);
+    }
+    else if (transform == Iceberg::PartitionTransform::Month)
+    {
+        return DateLUT::instance().lutIndexByMonthSinceEpochStartsZeroIndexing(static_cast<UInt32>(value));
+    }
+    else if (transform == Iceberg::PartitionTransform::Day)
+    {
+        return DateLUT::instance().getValues(static_cast<ExtendedDayNum>(value));
+    }
+    else if (transform == Iceberg::PartitionTransform::Hour)
+    {
+        DateLUTImpl::Values values = DateLUT::instance().getValues(static_cast<ExtendedDayNum>(value / 24));
+        values.date += (value % 24) * 3600;
+        return values;
+    }
+    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported partition transform for get day function: {}", transform);
+}
+
+Int64 getTime(Int32 value, Iceberg::PartitionTransform transform)
+{
+    DateLUTImpl::Values values = getValues(value, transform);
+    return values.date;
+}
+
+Int16 getDay(Int32 value, Iceberg::PartitionTransform transform)
+{
+    DateLUTImpl::Time got_time = getTime(value, transform);
+    return DateLUT::instance().toDayNum(got_time);
+}
+
+Range getPartitionRange(
+    Iceberg::PartitionTransform partition_transform, UInt32 index, ColumnPtr partition_column, DataTypePtr column_data_type)
```

Comment:
TBH I don't understand what this function does, can you clarify/add more comments?